tests/SleepForDays/SleepForDaysWorkflowTests.cs

```diff
@@ -0,0 +1,56 @@
+namespace TemporalioSamples.Tests.SleepForDays;
+
+using Temporalio.Activities;
+using Temporalio.Client;
+using Temporalio.Testing;
+using Temporalio.Worker;
+using TemporalioSamples.SleepForDays;
+using Xunit;
+using Xunit.Abstractions;
+
+public class SleepForDaysWorkflowTests : TestBase
+{
+    public SleepForDaysWorkflowTests(ITestOutputHelper output)
+        : base(output)
+    {
+    }
+
+    [Fact]
+    public async Task RunAsync_SleepForDays_Succeeds()
+    {
+        await using var env = await WorkflowEnvironment.StartTimeSkippingAsync();
+        var activities = new SleepForDaysActivities();
+
+        var activityExecutions = 0;
+        // Mock out the activity to assert number of executions
+        [Activity]
+        void SendEmail(string msg)
+        {
+            activityExecutions++;
+        }
+
+        using var worker = new TemporalWorker(
+            env.Client,
+            new TemporalWorkerOptions("sleep-for-days-task-queue").
+                AddActivity(SendEmail).
+                AddWorkflow<SleepForDaysWorkflow>());
+
+        var startTime = await env.GetCurrentTimeAsync();
+        await worker.ExecuteAsync(async () =>
+        {
+            var handle = await env.Client.StartWorkflowAsync(
+                (SleepForDaysWorkflow wf) => wf.RunAsync(),
+                new(id: $"wf-{Guid.NewGuid()}", taskQueue: worker.Options.TaskQueue!));
+            // Sleep for 90 days
+            await env.DelayAsync(TimeSpan.FromDays(90));
```

Comment:
Are you sure you're not time skipping before the workflow is even waiting for the number of days? You may be time skipping before the workflow even reaches the wait condition.

Reply:
I hadn't considered that. Is there a way to test for this? Maybe with a query?